magpie/api/webhooks.py

```diff
@@ -60,27 +73,81 @@ class WebhookAction(ExtendedEnum):
 
     CREATE_USER = "create_user"
     """
-    Triggered when a new user gets successfully created.
+    Triggered when a new :term:`User` gets successfully created.
 
     .. seealso::
         :ref:`webhook_user_create`
     """
 
     DELETE_USER = "delete_user"
-    """Triggered when an existing user gets successfully deleted.
+    """Triggered when an existing :term:`User` gets successfully deleted.
 
     .. seealso::
         :ref:`webhook_user_delete`
     """
 
     UPDATE_USER_STATUS = "update_user_status"
     """
-    Triggered when an existing user status gets successfully updated.
+    Triggered when an existing :term:`User` status gets successfully updated.
 
     .. seealso::
         :ref:`webhook_user_update_status`
     """
 
+    CREATE_USER_PERMISSION = "create_user_permission"
```

Comment:
Should we have an UPDATE_(USER|GROUP)_PERMISSION, or it's resolve as a delete then create operations?

Reply:
Yes its handled as 2 separate operations, so you can subscribe to only the need one(s).
I feel if we add an update one, we would have weird situations where you could have simultaneous create+update+delete configs, which would result in 4 webhook triggers.

Reply:
It's fine, I was only wondering if via an hypothetic update API call, we could mess a synched service.

Reply:
I think you are right, we can do a PUT request to replace any `access` or `scope` value for a target permission.
We can swap `allow` to `deny` or grant/remove lower level resources some permissions. 
So indeed, a third "on update" hook would be needed here.